### ISSUE TITLE
deprecation warning in the wxagg backend as of master

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#8718]
As of master, plotting something on the wxagg backend yields the warning
```
/home/antony/src/extern/matplotlib/lib/matplotlib/backends/backend_wx.py:1013: MatplotlibDeprecationWarning: The idle_event function was deprecated in version 2.1.
  FigureCanvasBase.idle_event(self, guiEvent=evt)
/home/antony/src/extern/matplotlib/lib/matplotlib/backend_bases.py:1960: MatplotlibDeprecationWarning: The IdleEvent class was deprecated in version 2.1.
  event = IdleEvent(s, self, guiEvent=guiEvent)
```

Of course we could choose to no longer call `self.Bind(wx.EVT_IDLE, self._onIdle)` (which ultimately triggers the warning) but if we decide to go with such a compatibility break then we may as well remove all of idle_event in 2.1 instead of going through a deprecation cycle...

**Comment by tacaswell** [comment]
Maybe the deprecation should go on subscription to the idle events not where they are?

**Comment by anntzer** [comment]
There's already one.
```
    def mpl_connect(self, s, func):
        <docstring>
        if s == 'idle_event':
            warn_deprecated(1.5,
                "idle_event is only implemented for the wx backend, and will "
                "be removed in matplotlib 2.1. Use the animations module "
                "instead.")

        return self.callbacks.connect(s, func)
```

Again, perhaps it is simpler to consider that that warning was sufficient and allow ourselves to entirely remove the idle_event machinery in 2.1.

**Comment by tacaswell** [comment]
You do make a compelling case.   Lets drop the `idle_event` call, but leave the class and def in place until 2.1 is out.

**Comment by anntzer** [comment]
Closed by #8722.
